packages/tx-construction/src/tx-builder/TxBuilder.ts

```diff
@@ -330,6 +501,101 @@ export class GenericTxBuilder implements TxBuilder {
     return this;
   }
 
+  async #computeCollateral(): Promise<{ collaterals: Set<Cardano.TxIn>; collateralReturn: Cardano.TxOut }> {
```

Comment:
TxBuilder class became pretty big. Please see if you can hoist some utils out of it without too much effort